### ISSUE TITLE
TypeError:  cannot assign to read only property 'name' function

### ISSUE THREAD

**Issue by miguelyoobic95** [m93a/filtrex#21]
Hi, 

I have recently upgraded the version of the filtrex package to `2.0.0` but our previously working code is now giving a `TypeError` as soon as the app loads.

The error seems to be come from the `jison.js` file in line 2143:
```
"push shift unshift forEach some every join sort".split(' ').forEach(function (e,i) {
    setMixin[e] = function () { return Array.prototype[e].apply(this._items, arguments); };
    setMixin[e].name = e;
});
```

The error stack is as follows:

```
Uncaught TypeError: Cannot assign to read only property ‘name’ of function ‘function () { return Array.prototype[e].apply(this._items, arguments); }’
   at index-0e1ff3fc.js:5286
   at Array.forEach (<anonymous>)
   at Object.factory [as jison/util/set] (index-0e1ff3fc.js:5284)
   at r (index-0e1ff3fc.js:3150)
   at Object.factory [as jison] (index-0e1ff3fc.js:3166)
   at r (index-0e1ff3fc.js:3150)
   at Module.../../../build/@shared/stencil/esm/index-0e1ff3fc.js (index-0e1ff3fc.js:6477)
   at __webpack_require__ (bootstrap:84)
   at Module.../../../build/@shared/stencil/esm/index-563ba7d7.js (index-563ba7d7.js:1)
   at __webpack_require__ (bootstrap:84)
```

Any help with this would be appreciated. If you need any further info, feel free to contact me and I will help with what I can.

**Comment by m93a** [comment]
Hello, sorry for inconvenience. The new version has got some breaking changes, including the call signature of `compileExpression`. Have you read [the changelog](https://github.com/m93a/filtrex/releases/tag/v2.0.0)? If you think your code is API-compilant, please post a minimal example of a breaking code. Thank you!

**Comment by vladimir-spintech** [comment]
Hi @m93a,

Thanks for maintaining this awesome project! 

I have also faced this problem, and it appears to be ES6 compatibility in [jison library ](https://github.com/m93a/filtrex/blob/master/src/lib/jison.js#L2145). From what I understand `name` property of a function is read-only. Maybe it is possible to implement one of the suggestions from this [thread](https://stackoverflow.com/questions/44288164/cannot-assign-to-read-only-property-name-of-object-object-object).

In the meantime I was able to use this library in my Angular project by changing typescript target to es5.

**Comment by m93a** [comment]
Hey,
Could you, please, provide an example code, so that I can reproduce the error and know, what I'm fixing? 😊

**Comment by vladimir-spintech** [comment]
For sure: https://github.com/vladimir-spintech/angular-filtrex-demo

[This line](https://github.com/vladimir-spintech/angular-filtrex-demo/blob/3cd6caa6c468fa9ab49be2d5655a737480ee3a2d/src/app/app.component.ts#L13) triggers the error and [this is](https://github.com/vladimir-spintech/angular-filtrex-demo/blob/3cd6caa6c468fa9ab49be2d5655a737480ee3a2d/tsconfig.json#L13) where you can change TS target to `es5` to make it work

**Comment by m93a** [comment]
Sorry for the delay, I have a lot of studying for uni. I'll look into it in February at the very least.

**Comment by m93a** [comment]
I published version 2.1.0-rc2. Please try it out in a real-world project and let me know if it works.

**Comment by agonzalez8** [comment]
@m93a It works, thank you.

**Comment by m93a** [comment]
I published it as stable 2.1.0 then!
